packages/ruleset/src/utils/check-subschemas-for-prop.js

```diff
@@ -1,36 +1,15 @@
-const checkSubschemasForProperty = (schema, name) => {
-  if (!schema) {
-    return false;
-  }
-
-  let propertyIsDefined = false;
+const checkCompositeSchemaForConstraint = require('./check-composite-schema-for-constraint');
+const isObject = require('./is-object');
 
-  // first check the properties
-  if (schema.properties) {
-    propertyIsDefined = name in schema.properties;
-  } else if (schema.oneOf || schema.anyOf) {
-    // every schema in a oneOf or anyOf must contain the property
-    const subschemas = schema.oneOf || schema.anyOf;
-    if (Array.isArray(subschemas)) {
-      propertyIsDefined = true;
-      for (const s of subschemas) {
-        if (!checkSubschemasForProperty(s, name)) {
-          propertyIsDefined = false;
-          break;
-        }
-      }
-    }
-  } else if (Array.isArray(schema.allOf)) {
-    // at least one schema in an allOf must contain the property
-    for (const s of schema.allOf) {
-      if (checkSubschemasForProperty(s, name)) {
-        propertyIsDefined = true;
-        break;
-      }
-    }
-  }
-
-  return propertyIsDefined;
+/**
+ * This function will return `true` if all possible variations of a (possibly) composite schema
+ * define a property with the specified name.
+ */
+const checkSubschemasForProperty = (schema, name) => {
```

Comment:
@dpopp07 @padamstx We might want to consider if this function is misnamed. Here, "subschemas" seems to mean something very different than what it means in `validateSubschemas()` which includes schemas embedded in `properties`, `items`, etc.

Reply:
That's true, they are checking different things, so I'm not opposed to a rename. `validateSubschemas()` does check all potential "schema" objects underneath a schema but this does not. I would support a more specific name that captures what this does better if you'd like to change it